modules/apps/portal-language/portal-language-override-web/src/main/java/com/liferay/portal/language/override/web/internal/portlet/action/ImportTranslationsMVCActionCommand.java

```diff
@@ -66,8 +61,7 @@ protected void doProcessAction(
 	}
 
 	private void _importTranslations(
-			ActionRequest actionRequest, File file, String languageId)
-		throws Exception {
+		ActionRequest actionRequest, File file, String languageId) {
```

Comment:
You could cut out the `File` middleman here and pass in an `InputStream` by using `UploadRequest.getFileAsStream` in the caller.

Reply:
But I check for the file name and extension here, it's this method's responsibility. Or am I missing some relevant info?

Reply:
does the name and extension matter? If the content is right, does the file have to have the right name?

Reply:
I think that enforces pattern, also it was already there :sweat_smile:

Reply:
yep. Thanks for your review btw

Reply:
anytime!